listener/pkg/metrics/metrics.go

```diff
@@ -0,0 +1,85 @@
+package metrics
+
+import (
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+const (
+	listenerRequestDuration            = "watcher_listener_request_duration"
+	listenerRequests                   = "watcher_listener_requests_total"
+	listenerRequestErrors              = "watcher_listener_request_errors_total"
+	listenerInflightRequests           = "watcher_listener_inflight_requests"
+	listenerExceedingSizeLimitRequests = "watcher_listener_exceeding_size_limit_requests_total"
+	listenerFailedVerificationRequests = "watcher_listener_failed_verification_requests_total"
+	requestURILabel                    = "request_uri_label"
+	listenerService                    = "listener"
+	serverNameLabel                    = "server_name"
+)
+
+var (
+	httpRequestDurationHistogram = prometheus.NewHistogramVec(prometheus.HistogramOpts{ //nolint:gochecknoglobals
```

Comment:
Just some questions, since Histogram is kind of expensive:
* How does this work without defining buckets?
* Can this metric be expressed with another simpler type?

Reply:
As agreed, will try the Gauge first on dev,  and if the data is not sufficient, we will use the histogram.